Add archive_all! and unarchive_all! methods

joshmenden opened the pull request:
Similar to the `destroy_all!` or `delete_all!` methods, we want to be able to call `archive_all!` and `unarchive_all!` on an ActiveRecord relation.

janxious commented on lib/expected_behavior/acts_as_archival_active_record_methods.rb:
```diff
@@ -36,5 +36,23 @@ def is_archival?
+        error_message = "The #{klass} must implement 'act_on_archivals' in order to call `archive_all!`"
+        raise NotImplementedError.new(error_message) unless archival?
+
+        each(&:archive!).tap { reset }
```
Just so we're clear, this will create n archive ids so that each archived item is its own tree. Is that intended, or do you want these all to count as a single archive action and have a shared archive id?

joshmenden replied:
Thanks for pointing that out -- I think in order to be consistent with the fact that parent-child archivals have the same archive id, let's go ahead and and have any record archived using `archive_all!` share the same archive_id. That change is currently in the PR. I've used the AR Relation's class and an integer representation of the time to generate the hash.

joshmenden commented:
@janxious -- thanks for the fast response.

I just squashed a few more commits into the PR.

- I've added a test file for testing basic `archive_all!` and `unarchive_all!` functionality
- I've changed `archive_all!` to archive each record in the association with the same Archive number.

joshmenden commented:
Also updated the README to document the new methods.

janxious commented:
Had some rubocop issues to iron out and a few env things since this hasn't been touched in a minute, so once we get those all settled we'll cut a new release, probs.

janxious commented:
@joshmenden had to make a number of changes to get this working and happy with rubocop and rails 5.2, and you can see those in the commit list on master if you're interested.
